File: Jim/src/sk/fiit/jim/garbage/plan/PlanTournamentWalk.java
package sk.fiit.jim.garbage.plan;

/* HIGHSKILL NEEDED: 0
 */

import sk.fiit.jim.LambdaCallable;
import sk.fiit.jim.agent.AgentInfo;
import sk.fiit.jim.agent.highskill.*;
import sk.fiit.jim.agent.models.AgentModel;
import sk.fiit.jim.agent.models.EnvironmentModel;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 * Rewrited PlanTournamentGetUpFromBelly from ruby
 * 
 * @author Adamik
 */

public class PlanTournamentWalk extends Plan {

	public static PlanTournamentWalk getInstance() {
		return new PlanTournamentWalk();
	}

	public void replan() {
		AgentModel agentModel = AgentModel.getInstance();

		if (EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("Beam");
			beamed = true;
			Vector3D start_position = Vector3D.cartesian(-5, 1, 0.4);
			Highskill_Queue.add(new Beam(start_position));
		} else if (agentModel.isOnGround() || agentModel.isLyingOnBack()
				|| agentModel.isLyingOnBelly()) {
			AgentInfo.logState("GetUp");
			Highskill_Queue.add(new GetUp());
		} else
			AgentInfo.logState("Go");
		Vector3D position_to_go = Vector3D.cartesian(10, 1, 0.4);
		Highskill_Queue.add(new GoToPosition(position_to_go,
				new LambdaCallable() {
					public boolean call() {
						return EnvironmentModel.beamablePlayMode();
					}
				}));

	}
}

File: Old_xml_and_scripts/Jim/plan/new-18.5.2013/planTournamentWalk.rb
### planTournamentWalk.rb ###
#	@author: A55Kickers		#
#############################

class PlanTournamentWalk < Plan
  
  def replan
    @plan.clear
    @agentInfo.whereIsBall();
    @agentInfo.isBallMine();
    @agentInfo.nearBall();
    @agentInfo.ballControlPosition();
    @agentInfo = Java::sk.fiit.jim.agent.AgentInfo.getInstance
	
    if EnvironmentModel.beamablePlayMode
    	@agentInfo.loguj('Beam');
    	@plan << Beam.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(-5, 1, 0.4))
    	@beamed = true
    elsif (me.on_ground? or me.is_lying_on_back? or me.is_lying_on_belly?)
    	@agentInfo.loguj('GetUp');
     	@plan << GetUp.new(Proc.new{me.on_ground?})
   	else
    	@agentInfo.loguj('Go');
    	@plan << GoToPosition.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(10, 1, 0.4), Proc.new{not EnvironmentModel.beamablePlayMode})
    end
  end
end